scripts/widgets/mz-search-pagination-dd.js

```diff
@@ -0,0 +1,168 @@
+        var timeout = null;
+        var pageIndex = 0;
+        var lastPageIndex = 0;
+        var filteredVal = false;
+        var dataSet = new options.model({ pageSize: options.pageSize });        
+        dataSet.apiGet().then(function (accounts) {            
+            lastPageIndex = accounts.data.pageCount;
+            var divElm = "";
+            _.each(accounts, function (data, i) {
+                if (!_.isUndefined(data.data[options.displayField])) {
+                    divElm += '<a class="mozu-dropdown" data-id="'+ data.data.id+'" tabindex="-1" href="javascript:void(0)">' + data.data[options.displayField] + '</a>'; 
+                }     
+            });
+            $(self.selector).append('<div tabindex="-1" class="listData">' + divElm + '</div>');
+            $('<div class="mz-pagination-url"><a class="firstPage" data-page-val="0" href="javascript:void(0)" style="padding: 0 10px;"><<</a> <a class="prevPage" href="javascript:void(0)" style="padding: 0 10px;"><</a>  <span class="displayPageCount" style="padding: 0 10px;"> <span clss="count_val">' + (accounts.data.startIndex + 1) + '</span> of ' + lastPageIndex + ' </span> <a class="nextPage" data-pagination="0" href="javascript:void(0)" style="padding: 0 10px;">></a> <a class="lastPage" href="javascript:void(0)" style="padding: 0 10px;">>></a><div class="mz-dd-loading"></div></div>').appendTo(self.selector);     
+
+            $(document).on('click', '.mz-dd-open', function () {
+                $('.dropdown_data').hide();
+                $(this).removeClass('mz-dd-open').addClass('mz-dd-close');
+            });
+
+            $(document).on('click', '.mz-dd-close', function () {
+                $('.dropdown_data').show();
+                $(this).removeClass('mz-dd-close').addClass('mz-dd-open');
+            });
+
+            // API call for pagination click
+            function paginationLoad(pageNumber) {
+                $('.mz-dd-loading').show();
+                var inputValue = $('.mz-dd-search').val();
+                var dataSet = new options.model({ startIndex: options.pageSize * pageNumber, pageSize: options.pageSize, isActive: true });
+                if (filteredVal) {
+                    dataSet = new options.model({ startIndex: options.pageSize * pageNumber, pageSize: options.pageSize, filter: options.displayField + ' ' + options.filterOption + ' "' + inputValue + '"', isActive: true });
+                }
+                dataSet.apiGet().then(function (accounts) {   
+                    $('.listData').text('');
+                    _.each(accounts, function (data) {
+                        if (!_.isUndefined(data.data[options.displayField])) {
+                            $('.listData').append('<a class="mozu-dropdown" data-id="' + data.data.id +'" tabindex="-1" href="javascript:void(0)">' + data.data[options.displayField] + '</a>');
+                        }
+                    });
+                    $('.mz-dd-loading').hide();
```

Comment:
Ideally all constants like this would be defined at the top of the file so you can reuse the constant in multiple places. I'll let it slide this time but in the future please use constants for such things.

Reply:
Sure, Thanks